### ISSUE TITLE
Feature Request: Direct Message when a Review is Requested

### ISSUE THREAD

**Issue by philschatz** [integrations/slack#382]
[Review Requests](https://github.com/blog/2291-introducing-review-requests) are a great way to get feedback and I use them daily but they quickly get lost/deleted in my email.

Currently, I use slackbot to remind me to check my "Reviews Requested" page but it would be awesome if the GitHub app sent a Direct Message whenever a review was requested and then updated it once I reviewed.

The command I currently use is `/remind me "Check https://github.com/pulls?q=is%3Aopen+review-requested%3Aphilschatz" every day`

**Comment by ryandeussing** [comment]
This should be configurable, e.g. users should be able to determine if they are notified in Slack for review requests, and how (message, message in a specific/special channel, or DM).

**Comment by LandonSchropp** [comment]
This would be awesome! Specifically, my team would like to have the following:

* Whenever someone is added to a PR, the GitHub Slack bot tags them in a message.
* If a PR hasn't been reviewed after a certain amount of time (usually because somebody forgot), then it sends them a reminder message.

**Comment by thealexwilson** [comment]
Yes, please.

**Comment by buzinas** [comment]
Also, it'd be great to have the "Reviewers" section of an open PR just below the "Assignees" one:

![image](https://user-images.githubusercontent.com/7298695/38431330-43444c14-399a-11e8-902e-bdb8d1ccd6df.png)

**Comment by abhishek1nair** [comment]
Github explicitly has this [event](https://developer.github.com/v3/activity/events/types/#pullrequestevent) but I wonder why this has not been implemented. Even if the channel the bot posts to gets an update saying that review has been requested on certain PR it would be really helpful.

**Comment by vintagepenguin** [comment]
We were also interested in this functionality, but at the same time we started using [Pull Reminders](http://pullreminders.com/) from the [GitHub Marketplace](https://github.com/marketplace/pull-reminders/).

P-Reminders does DMs for review requests, which is cool but it was the periodic reminders for our team that we we really needed. Some guys on our team need consistent reminders instead of 1, and we wanted to control that, as well as having some of the reporting features.

**Comment by hanzki** [comment]
In our team, we use "request a review" in a way that people self-request themselves to pull requests. We have a slack channel for noticing opening of new pull requests and the one who decides to do the review gives others heads up there in order to avoid too many people reviewing the same PR at the same time. We would love to have the slack integration be able to show on the channel whenever someone adds themselves as a reviewer to a PR as that would automatically inform everyone.

**Comment by EnixCoda** [comment]
I implemented one on my own. And I've made its setup steps needed as simple as possible.

https://github.com/EnixCoda/GitHubReviewerNotificationForSlack

It makes my life much easier!

**Comment by sebbean** [comment]
seems like such a great thing to have in this integration!

**Comment by abinoda** [comment]
@sebbean Check out http://github.com/marketplace/pull-reminders. I'd be happy to give your team a free account.

**Comment by sebbean** [comment]
@abinoda i'm using it! it's great!
thank you so much.

is there a way to get reminders for when a PR has been approved?
seems like the last piece of the workflow puzzle.

**Comment by abinoda** [comment]
@sebbean Hmm. If you go under "My DM Settings" there should be an option for getting a DM when your PR is approved (also, don't want to clog up this issue with discussion so feel free to open one here: http://github.com/pullreminders/backlog/issues)

**Comment by take** [comment]
any chance of this request being implemented by the officials? :eyes:

**Comment by guillaume-eb** [comment]
GitHub has acquired Pull Panda, the makers of the Pull Reminders app discussed above: https://github.blog/2019-06-17-github-acquires-pull-panda/

"_ We plan to integrate these features into GitHub but hope you’ll start benefiting from them right away_"

**Comment by liamhession** [comment]
Should this be marked as closed now? I almost navigated away after Googling and ending up here because i assumed "Open" meant this wasn't yet possible. But then i found Pull Panda in the thread here

**Comment by atribbot** [comment]
👍

**Comment by scepticulous** [comment]
Thanks, everybody for your awesome input.

It is likely that we are going to add some functionality related to this to the official Slack integration independent of the pull reminders integration.

I have a quick question to everybody involved: Is this functionality required to be a "DM" or is anybody also looking for PR assignments in the regular channel subscription.
This the core idea of this features seems "assigned to me" it seems as if DM is the core idea. 
Please let me know if there are other use-cases.

`cc` @philschatz @philschatz @ryandeussing for DM feedback

`cc` @LandonSchropp  since your idea *sounds* more to be around "sending the assignment event to the channel and '@'-mention the user that is assigned directly or through a team membership".

Adding "reminders" is currently out of scope for the official Slack integration.

**Comment by tatimblin** [comment]
> anybody also looking for PR assignments in the regular channel subscription.

We use a pr channel per team. My first thought when imagining automating the process was a reply @-ing the reviewer to the original post.

![](https://i.gyazo.com/c866c30ce6a8938036e2c033fa28bfd2.png)

If it were DM only to the reviewer it would still accomplish the goal, just change our process a little.

**Comment by LandonSchropp** [comment]
Since I posted that message, we've started using Pull Reminders, which does exactly what we needed. When someone is assigned a PR review, it sends them a DM directly letting them know.

**Comment by kkozmic** [comment]
> Is this functionality required to be a "DM" or is anybody also looking for PR assignments in the regular channel subscription.

For my team either one is fine. We want to bring the fact that someone was requested as a reviewer to their attention on Slack - we're not too fussed about how it gets delivered, so whatever is easier to implement we'd be happy with @scepticulous

**Comment by christhekeele** [comment]
> Is this functionality required to be a "DM" or is anybody also looking for PR assignments in the regular channel subscription.

I prefer the channel-orientated vector (allowing for round-robin, pick-up PRs within specific teams) but not too put off if that's not available out of the gates

**Comment by sregg** [comment]
It seems that Pull Reminders/Pull Panda have been removed from the market place, is there any update the having this feature in the official app, or any new alternatives?

**Comment by gerbus** [comment]
Yeah. Do I now have to wait until this has been integrated with Github? Is there a timeline estimate?

**Comment by lynncyrin** [comment]
ping @dennissivia @abinoda this feature is very high value for our workflows at work, can we please get an update on what's going on here? I see there's a PR implementing this feature here => https://github.com/integrations/slack/pull/861

**Comment by lynncyrin** [comment]
@tebriel @wilhelmklopp @jules2689

> this feature is very high value for our workflows at work, can we please get an update on what's going on here? I see there's a PR implementing this feature here => #861

**Comment by Kaylee-Cole** [comment]
That's good

**Comment by tekstrand** [comment]
Would love to see this feature. A great alternative to having to subscribe to a whole repo where hundreds of committers across multiple teams are contributing. The noise would be too much. This feature would solve that!

**Comment by sregg** [comment]
We ended up using Toast for our team:
https://slack.com/apps/AC6M092SD-toast

Works great and supports direct messages to requested reviewers among other features.

**Comment by willsmythe** [comment]
The plan is to add this and many of the other Pull Panda DM features to GitHub in the next month or two. Once available, you'll be able to configure them via your GitHub settings page. Watch the [GitHub blog](https://github.blog/) for updates.

**Comment by lynncyrin** [comment]
Thanks so much for the info @willsmythe! It would have been nice to have gotten an update on this earlier, but I super appreciate you letting us know now 🙏

**Comment by fej-snikduj** [comment]
Any update on this @willsmythe?

**Comment by tebriel** [comment]
https://github.blog/changelog/2019-11-14-scheduled-reminders-beta you can sign up the beta is open!

**Comment by kkozmic** [comment]
> https://github.blog/changelog/2019-11-14-scheduled-reminders-beta you can sign up the beta is open!

This looks to be missing two basic features (yes, I know it's a preview):
* per repo/per channel settings (rather than per user/team), so I can say "for repo x and y if a PR review is requested DM the appropriate user" rather than every single person having to do that for themselves
* timezone aware daily reminders - my team is globally distributed and not being able to remind users at, say 9am *their* time is a drawback

**Comment by tebriel** [comment]
@kkozmic thanks, I know it's not perfect. I'm taking a note of your feedback and sharing it with Product.

**Comment by freakyfelt** [comment]
My team uses the DM upon review requested/review approved feature in Pull Panda and it seems that today they are blocking new signups. Digging through the official integration docs I still don't see this as an option. Am I correct in that assumption?

**Comment by ashafer01** [comment]
Similar to this, I'd like to be able to map an org teams to the Slack channel for those team members, and have review requests sent to that channel.

**Comment by yoadfe** [comment]
How can we enable real-time alerts to all of the team ?
instead of having each team members enable it for himself.

**Comment by tebriel** [comment]
@ashafer01 You can have team-based scheduled reminders, I think [this document](https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-scheduled-reminders-for-your-team) describes what you're looking for.

@yoadfe realtime reminders are currently only available for individual users to configure. I'll pass the idea of enabling realtime for an entire group of people on to product.

**Comment by obahareth** [comment]
> Pull Assigner and Pull Reminders will be retired on October 14, 2020. Migrate to code review assignment and scheduled reminders to continue using these tools.

This feature is the primary reason we use PullReminders, we don't use anything scheduled, just regular reminders in DMs from PullReminders, but we're being asked to migrate to Scheduled Reminders without an alternative for this feature. Could we please extend the deadline for retiring Pull Reminders until this feature is added to this integration?

**Comment by tebriel** [comment]
@obahareth You can receive direct message reminders from "scheduled reminders" (there are live reminders, we call them 'personal reminders') in your user setting: https://github.com/settings/reminders. If you `Enable real-time alerts` you can get direct messages about PRs assigned to you. Is this what you're asking for?
![Personal Reminders Settings](https://user-images.githubusercontent.com/821688/87945558-e840de00-ca6e-11ea-89a5-4ee02d1b9a5d.png)

**Comment by obahareth** [comment]
@tebriel This works but from my attempts/understanding I have to ask every member to enable it for themselves right? If my understanding is correct, is a little different than Pull Reminders where I could set the default setting for the whole organization.

**Comment by tebriel** [comment]
@obahareth yes, it is different from the setup in Pull Reminders. I've forwarded your feedback to the product lead for scheduled reminders.

**Comment by EtienneM** [comment]
I encountered issues a few times with this feature where my team mate never received the notification asking him for the review. I also never received the notification when he merged the PR. I can send the PR in DM if needed to help debug the issue.

**Comment by zaverden** [comment]
I cannot set up scheduled reminders for my case. May be somebody has an advice for me?

I'm in outsourcing team and we were added to `github.com/client-company` to work in some private repos. 

I cannot connect `github.com/client-company` to `my-team.slack.com` workspace, because nobody will grant me this high permission (because of clear security concerns).

And client's admins cannot do this because they are not a part of `my-team.slack.com`.

What is a expected way to set up reminder in this case? Am I missing some documentation on it?

**Comment by tebriel** [comment]
@zaverden I don't believe scheduled reminders supports this use case, could you open an issue at support.github.com to track that concern? 🙇🏻

**Comment by C-E-Rios** [comment]
It's a shame that the best features from the original Pull Reminders slack integration have gotten lost in the transition to GitHub. As many others have said, allowing for team members to receive a private message on Slack when they've been added as a reviewer to a pull request is super useful! Pull reminders gave us this in a really straight forward way, with minimal setup. Can we have this back please?

**Comment by ashokirla** [comment]
@C-E-Rios We understand your concern. We are currently looking at improving personal notifications functionality. This includes when you are participating or assigned for prs, issues, releases and other activities in GitHub. 
Can we have a quick sync up call. We would like to understand your scenarios and share our plans for personal notifications story.

**Comment by ashokirla** [comment]
> I cannot set up scheduled reminders for my case. May be somebody has an advice for me?
> 
> I'm in outsourcing team and we were added to `github.com/client-company` to work in some private repos.
> 
> I cannot connect `github.com/client-company` to `my-team.slack.com` workspace, because nobody will grant me this high permission in `github.com/client-company` organization (because of clear security concerns).
> 
> And client's admins cannot do this because they are not a part of `my-team.slack.com`.
> 
> What is a expected way to set up reminder in this case? Am I missing some documentation on it?

@zaverden Are you able to use our GitHub app in slack and subscribe to client-company org and get notifications. 
The prerequisites for schedule reminders are github app installed in slack, authorize slack to use repositories in your org. This is a one-time activity that can be done by org admin. After this you should be able to configure schedules on the slack workspace.

I am happy to get over a call discuss in detail.

**Comment by zaverden** [comment]
@ashokirla I cannot authorize my slack workspace in github org, because I don't have required permissions in client's github org. 
![image](https://user-images.githubusercontent.com/13230284/95965242-e26f5480-0e33-11eb-9338-26cdce8b6191.png)

What will happen if github user who connected slack workspace is no longer admin in github org? does workspace  continue to be connected? I may ask for temporal permission just to connect slack and to be removed right after that.

**Comment by ashokirla** [comment]
@zaverden 
Yes, it is a one time thing. You can ask your org admin to authorize slack on github or you can get permissions to configure it, get the create the schedule reminder and then get your access removed. However, if you want to add a new reminder the org settings page, you need to be admin on the org. 

Are you configuring reminders for your personal app or for a channel. If you are doing it at a channel level, for an organization,  we have Team level schedule reminders where you only need to be a Team admin. 
![image](https://user-images.githubusercontent.com/25578249/95983064-a4c3f900-0e3e-11eb-9b58-9028967c8ed4.png)

**Comment by Restuta** [comment]
There is an alternative https://toast.ninja/, we've been featured in Slack marketplace several times (and featured right now https://slack.com/apps) and now provide GitHub analytics right in Slack on top of advanced notifications and all that. 

Toast supports onboarding based on GitHub teams and default configuration for all teammates. We care about design and we care about the noise! Our team has an internal alert when somebody is over-notified and we review and improve based on those use-cases, continuosly investing in the core product. 

Aside from that Toast supports App Home, which is your personal review board and advanced Slack commands. We help everybody to not only review their own PRs, but also help unblock teammates and stay aware of what is going on in the team.

cc @obahareth

**Comment by ssyberg** [comment]
Wanted to add another use case here - as a eng manager, I like to know when a dev grabs a new issue off the queue, getting a slack alert of really any change to a project would be very useful!

**Comment by nickfiacco-bolt** [comment]
If I'm part of multiple teams, I can't configure real-time notifications to only send me notifications about a single team.

**Comment by rattrayalex** [comment]
This seems to work for me now! [This comment](https://github.com/integrations/slack/issues/382#issuecomment-661054689) shows you how; go to https://github.com/settings/reminders , click "Enable real-time alerts", and then "You are assigned a review". I get a slack DM from the Github app when someone clicks to "Request review" from me 🎉

**Comment by edkimmel** [comment]
Any update on this feature?  We are still unable to get real time alerts sent to a designated slack channel when a review is requested for a specific team.  Right now, the integration only works on open, close, and re-open.

**Comment by noalbalint** [comment]
> @obahareth You can receive direct message reminders from "scheduled reminders" (there are live reminders, we call them 'personal reminders') in your user setting: https://github.com/settings/reminders. If you `Enable real-time alerts` you can get direct messages about PRs assigned to you. Is this what you're asking for? ![Personal Reminders Settings](https://user-images.githubusercontent.com/821688/87945558-e840de00-ca6e-11ea-89a5-4ee02d1b9a5d.png)

@tebriel is there another step required for this to work? I have GitHub app in Slack and have selected all these settings, but still not getting any notifications :/

**Comment by tebriel** [comment]
I'm not sure, I haven't worked on that team in a few years, you'll probably want to open a [support ticket](https://support.github.com/)

**Comment by benmayne** [comment]
so as of April 2024, the real time alerts mentioned above are working for me.

**Comment by valentine-morozov** [comment]
It's not working for me too, I've linked Slack and Github but it does nothing

**Comment by sarthakydv** [comment]
Working well for me
